torchmetrics/regression/tweedie_deviance.py

```diff
@@ -0,0 +1,119 @@
+from torchmetrics.utilities.data import dim_zero_cat
+
+
+class TweedieDevianceScore(Metric):
+    r"""
+    Computes the `Deviance Score <https://en.wikipedia.org/wiki/Tweedie_distribution#The_Tweedie_deviance>`_ between
+    targets and predictions:
+
+    .. math::
+        deviance\_score(\hat{y},y) =
+        \begin{cases}
+        (\hat{y} - y)^2, & \text{for }power=0\\
+        2 * (y * log(\frac{y}{\hat{y}}) + \hat{y} - y),  & \text{for }power=1\\
+        2 * (log(\frac{\hat{y}}{y}) + \frac{y}{\hat{y}} - 1),  & \text{for }power=2\\
+        2 * (\frac{(max(y,0))^{2}}{(1 - power)(2 - power)} - \frac{y(\hat{y})^{1 - power}}{1 - power} + \frac{(\hat{y})
+            ^{2 - power}}{2 - power}), & \text{otherwise}
+        \end{cases}
+
+    where :math:`y` is a tensor of targets values, and :math:`\hat{y}` is a tensor of predictions.
+
+    Forward accepts
+
+    - ``preds`` (float tensor): ``(N,)``
+    - ``targets`` (float tensor): ``(N,)``
+
+    Args:
+        power:
+            - power = 0 : Normal distribution. (Requires: y_true and y_pred can be any real numbers.)
+            - power = 1 : Poisson distribution. (Requires: y_true >= 0 and y_pred > 0.)
+            - power = 2 : Gamma distribution. (Requires: y_true > 0 and y_pred > 0.)
+        compute_on_step:
+            Forward only calls ``update()`` and return ``None`` if this is set to ``False``.
+        dist_sync_on_step:
+            Synchronize metric state across processes at each ``forward()``
+            before returning the value at the step.
+        process_group:
+            Specify the process group on which synchronization is called.
+            default: ``None`` (which selects the entire world)
+        dist_sync_fn:
+            Callback that performs the allgather operation on the metric state. When ``None``, DDP
+            will be used to perform the all gather.
+
+    Example:
+        >>> from torchmetrics import TweedieDevianceScore
+        >>> targets = torch.tensor([1.0, 2.0, 3.0, 4.0])
+        >>> preds = torch.tensor([4.0, 3.0, 2.0, 1.0])
+        >>> deviance_score = TweedieDevianceScore(power=0)
+        >>> deviance_score(preds, targets)
+        tensor(5.)
+
+    """
+    preds: List[Tensor]
+    targets: List[Tensor]
+
+    def __init__(
+        self,
+        power: int = 0,
+        compute_on_step: bool = True,
+        dist_sync_on_step: bool = False,
+        process_group: Optional[Any] = None,
+        dist_sync_fn: Callable = None,
+    ) -> None:
+        super().__init__(
+            compute_on_step=compute_on_step,
+            dist_sync_on_step=dist_sync_on_step,
+            process_group=process_group,
+            dist_sync_fn=dist_sync_fn,
+        )
+        if power < 1 and power > 0:
+            raise ValueError(f"Deviance Score is not defined for power={power}.")
+
+        self.power: int = power
+
+        self.add_state("preds", [], dist_reduce_fx="cat")
+        self.add_state("targets", [], dist_reduce_fx="cat")
```

Comment:
this is more memory efficient
```suggestion
        self.add_state("sum_deviance_score", torch.zeros(1), dist_reduce_fx="sum")
        self.add_state("num_obs", torch.zeros(1), dist_reduce_fx="sum")
```